airflow/contrib/hooks/gcs_hook.py

```diff
@@ -172,7 +175,8 @@ def download(self, bucket, object, filename=None):
         return downloaded_file_bytes
 
     # pylint:disable=redefined-builtin
-    def upload(self, bucket, object, filename, mime_type='application/octet-stream'):
+    def upload(self, bucket, object, filename,
+               gzip=False, mime_type='application/octet-stream'):
```

Comment:
One last comment, can you move the `gzip` argument to the very end? Otherwise backward compatibility will be broken.